lib/chat/chat_messenger.dart

```diff
@@ -37,24 +37,87 @@ class ChatMessengerState extends State<ChatMessenger> {
 
   Store<AppState> store;
 
+  List<DocumentSnapshot> _messagesSnapshots;
+  bool _isLoading = false;
+  Message lastMessage;
   String recipient;
 
   File imageFile;
   String imageUrl;
 
   final TextEditingController textInputController = TextEditingController();
+  final ScrollController scrollController = ScrollController();
   final ImagePicker imagePicker = ImagePicker();
 
+  @override
+  void setState(fn) {
+    if (mounted) {
+      super.setState(fn);
+    }
+  }
+
   @override
   void initState() {
     super.initState();
     imageUrl = '';
+
+    scrollController.addListener(() {
+      if (scrollController.position.atEdge) {
+        if (scrollController.position.pixels != 0) {
```

Comment:
IDK if it's just my eyes not keep tracking of brackets, but it looks like both conditions are required in this case and can be compressed to an `&&`?